source/common/http/http1/codec_impl.cc

```diff
@@ -469,8 +469,23 @@ int ConnectionImpl::onHeadersCompleteBase() {
     protocol_ = Protocol::Http10;
   }
   if (Utility::isUpgrade(*current_header_map_)) {
-    ENVOY_CONN_LOG(trace, "codec entering upgrade mode.", connection_);
-    handling_upgrade_ = true;
+    // Ignore h2c upgrade requests until we support them.
+    // See https://github.com/envoyproxy/envoy/issues/7161 for details.
+    if (current_header_map_->Upgrade() &&
+        absl::EqualsIgnoreCase(current_header_map_->Upgrade()->value().getStringView(),
+                               Http::Headers::get().UpgradeValues.H2c)) {
+      ENVOY_CONN_LOG(trace, "removing unsupported h2c upgrade headers.", connection_);
+      current_header_map_->removeUpgrade();
```

Comment:
Do you think we should strip the "HTTP2-Settings" header also? I'm not sure it matters but just throwing that out there for consideration (if we do decide to strip it I would not add it to the O(1) header map because this is such a corner case).

Reply:
Done.  Talked to @PiotrSikora and he confirmed that this is the last one.

Reply:
Removed from O(1) header map.